be/tool/beck.c

```diff
@@ -1107,13 +1125,19 @@ static int seghdr_ver(struct scanner *s, struct rectype *r, char *buf)
 static void genadd(uint64_t gen)
 {
 	int i;
+	if (gen_count >= ARRAY_SIZE(g))
+		return;
 
-	for (i = 0; i < ARRAY_SIZE(g); ++i) {
-		if (g[i].g_gen == gen || g[i].g_count == 0) {
+	for (i = 0; i < gen_count; ++i) {
+		if (g[i].g_gen == gen) {
 			g[i].g_count++;
 			break;
 		}
 	}
+	if (i == gen_count) {
+		g[gen_count].g_gen = gen;
+		g[gen_count++].g_count = 1;
```

Comment:
It's better not to use the self increment operation here:
It is error prone. 

Let's use a separate clause:
```
g[gen_count].g_count = 1;
gen_count++;

```

Reply:
Done.